pkg/patterns/declarative/pkg/kubectlcmd/client.go

```diff
@@ -49,7 +56,8 @@ func (console) Run(c *exec.Cmd) error {
 }
 
 // Apply runs the kubectl apply with the provided manifest argument
-func (c *Client) Apply(ctx context.Context, namespace string, manifest string, validate bool, extraArgs ...string) error {
+func (c *Client) ApplyAlt(ctx context.Context, namespace string, manifest string, validate bool,
```

Comment:
Nit: I think having two types might be clearer here - e.g. one `ExecKubectl` and one called `DirectApply`, that both implement the same interface.  We can discuss in our GSoC sync :-)